test/OpenTelemetry.Contrib.Instrumentation.Quartz.Tests/QuartzDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,397 @@
+
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+            using var tel = Sdk.CreateTracerProviderBuilder()
+                .SetSampler(new AlwaysOnSampler())
+                .AddQuartzInstrumentation()
+                .AddProcessor(activityProcessor.Object)
+                .Build();
+            var schedulerConfig = SchedulerBuilder.Create("AUTO", "Scheduler");
+            schedulerConfig.UseDefaultThreadPool(x => x.MaxConcurrency = 10);
+            var scheduler = await schedulerConfig.BuildScheduler();
+
+            scheduler.Context.Put("BARRIER", barrier);
+            scheduler.Context.Put("DATESTAMPS", jobExecTimestamps);
+            await scheduler.Start();
+
+            JobDataMap jobDataMap = new JobDataMap { { "A", "B" } };
+
+            var name = Guid.NewGuid().ToString();
+            var job = JobBuilder.Create<TestJob>()
+                .WithIdentity(name, SchedulerConstants.DefaultGroup)
+                .UsingJobData(jobDataMap)
+                .Build();
+
+            var trigger = TriggerBuilder.Create()
+                .WithIdentity(name, SchedulerConstants.DefaultGroup)
+                .StartNow()
+                .Build();
+
+            // Act
+            await scheduler.ScheduleJob(job, trigger);
+
+            barrier.SignalAndWait(TestTimeout);
+
+            await scheduler.Shutdown(true);
+
+            // Assert
+            Assert.Equal(3, activityProcessor.Invocations.Count);
+            var activity = (Activity)activityProcessor.Invocations[1].Arguments[0];
+
+            Assert.Contains("execute ", activity.DisplayName);
+            Assert.Equal("Quartz.Job.Execute", activity.OperationName);
+            Assert.Equal(ActivityKind.Internal, activity.Kind);
+            Assert.Equal("Scheduler", activity.Tags.SingleOrDefault(t => t.Key.Equals("scheduler.name")).Value);
+            Assert.Equal(SchedulerConstants.DefaultGroup, activity.Tags.SingleOrDefault(t => t.Key.Equals("job.group")).Value);
+            Assert.Equal(SchedulerConstants.DefaultGroup, activity.Tags.SingleOrDefault(t => t.Key.Equals("trigger.group")).Value);
+        }
+
+        [Fact]
+        public async Task Should_Create_Activity_And_Enrich_When_Enrich()
+        {
+            // Arrange
+            Barrier barrier = new Barrier(2);
+            List<DateTime> jobExecTimestamps = new List<DateTime>();
+
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+            var jobDetailsPropertyFetcher = new PropertyFetcher<object>("JobDetail");
+            var jobDataMapPropertyFetcher = new PropertyFetcher<object>("JobDataMap");
+            using var tel = Sdk.CreateTracerProviderBuilder()
+                .SetSampler(new AlwaysOnSampler())
+                .AddQuartzInstrumentation(q =>
+                    q.Enrich = (a, s, p) =>
+                    {
+                        if (jobDetailsPropertyFetcher.TryFetch(p, out var jobDetail))
```

Comment:
We should be doing this fetching ourselves, and not require users to do it.

Reply:
fetched internally, updated  README to reflect changes to be inline with other implementations